Run tests against multiple rails versions on GitHub actions

mjobin-mdsol opened the pull request:
I noticed Travis does not run tests on this project for an unknown reason... 

And since this gem is public, I thought we could use Github Actions,

I setup several Gemfile for rails 4.2 -> 6.1

Note that tests only pass if we lock rspec to 2.x
Otherwise, we are stubbing a dead method, so every rails versions fails

https://mdsol.slack.com/archives/C8JH3MH17/p1650857924277109?thread_ts=1650464131.151619&cid=C8JH3MH17

jfeltesse-mdsol commented:
Travis does run but the last few weren't run because it required a PR to be opened (build branches was disabled -- I inverted those settings to it builds branches instead of PRs).

So I think for now we should keep travis and maybe at a later time use GHA.
Besides, I think we can drop support for rails `< 5.2`.

jfeltesse-mdsol commented on gemfiles/rails_4.2.gemfile.lock:
```diff
@@ -0,0 +1,214 @@
+PATH
```
can you add appraisal as a dev dependency and add `/gemfiles/*.gemfile.lock` to the gitignore?

mjobin-mdsol replied:
.lock is required with rails 4.2 for github actions to automatically pick up bundler 1.x
we can remove when we remove rails 4.2
that's the best solution I found using github actions on many open source projects I maintain

jfeltesse-mdsol commented on rubycas-client.gemspec:
```diff
@@ -11,20 +11,27 @@ Gem::Specification.new do |s|
+  # 2.1 renamed get_session in 2019 -- https://github.com/redis-store/redis-rack/commit/41b3083d9a054ad28dd1151d5ad30c9a0277a792
+  s.add_runtime_dependency 'redis-rack', '< 2.1'
+
+  # Optional Dependencies Dalli or Redis
```
I guess we need to have dalli as a dev dep as well then.

mjobin-mdsol replied:
I thought it would be a breaking changes. so I didn't move it. but sure I can move. let me know.

redis* were in the Gemfile

jfeltesse-mdsol replied:
fair enough, good to keep as is for now.

mjobin-mdsol commented:
> Travis does run but the last few weren't run because it required a PR to be opened (build branches was disabled -- I inverted those settings so it builds branches instead of PRs).

I see ... I rebased and removed the removal of the travis file.
 
> So I think for now we should keep travis and maybe at a later time use GHA. Besides, I think we can drop support for rails `< 5.2`.

Should we not wait until imedidata's upgrades is  running ?

jfeltesse-mdsol commented:
> Should we not wait until imedidata's upgrades is running ?

No, because it's running just fine with its current version of rails and this gem. Doubtful any last minute bugfix would crawl in and even if it did, we could hack on a specific branch and use that in imed.

jfeltesse-mdsol commented on spec/support/active_record_helpers.rb:
```diff
@@ -19,7 +19,8 @@ def teardown_active_record
   end
 
-  class RubyCasTables < ActiveRecord::Migration
+  migration_base_class = Rails::VERSION::STRING.to_f >= 5.2 ? ActiveRecord::Migration[4.2] : ActiveRecord::Migration
```
maybe we don't need that check anymore

mjobin-mdsol commented:
closing in favor of #64 

if there are any interests seeing the CI execute in 15s instead of 5 minutes, let me know, the github action recipe is ready...
